### PR TITLE
use meaningful constants instead of senseless numeric literals

### PR DESCRIPTION
**Pull request by spirit-q2**
also fix for game logic and some comments

### CODE REVIEW

File: gameEngine/app.php [hunk 1]
```diff
@@ -44,7 +44,7 @@
 if (!empty($_POST['action']) && 'update' == $_POST['action']) {
     $gameHandler->updatePing($_POST['room']);
     $gameHandler->removeInactivePlayer($_POST['room']);
```
Comment (spirit-q2) [inline gameEngine/app.php:46]: Your session is not available here (line 46), yet you are trying to access `$_SESSION['score']` in the method `removeInactivePlayer()`. You must start session **before** accessing any session variables.

File: gameEngine/app.php [hunk 2]
```diff
@@ -44,7 +44,7 @@
...
-    $allMoves = $gameHandler->getMoves();
+    $allMoves = $gameHandler->getMoves(); // your session starts here
     $dataSet = ['playerDetails' => $gameHandler->getPlayersNames($_POST['room']), 'movesMade' => $allMoves, 'winner' => $gameHandler->whoIsWinning($allMoves)];
```
Comment (spirit-q2) [inline gameEngine/app.php]: I understand that you need to start session in the `getMoves()` method, accounting your game logic. But you also need to start it in the `removeInactivePlayer()` ;)
To eliminate php's complaining about calling `session_start()` twice in a row (php's throwing a warning about that) you can use the trick:

    if (PHP_SESSION_ACTIVE != session_status()) {
        session_start();
    }

File: gameEngine/app.php [hunk 3]
```diff
@@ -44,7 +44,7 @@
 if (!empty($_POST['action']) && 'update' == $_POST['action']) {
     $gameHandler->updatePing($_POST['room']);
     $gameHandler->removeInactivePlayer($_POST['room']);
-    $allMoves = $gameHandler->getMoves();
+    $allMoves = $gameHandler->getMoves(); // your session starts here
```
Comment (spirit-q2) [inline gameEngine/app.php:47]: I understand that you need to start session in the `getMoves()` method, accounting your game logic. But you also need to start it in the `removeInactivePlayer()` ;)
`php` complains about calling `session_start()` twice in a row (it throws a warning about that). to eliminate warning you can use the trick:

    if (PHP_SESSION_ACTIVE != session_status()) {
        session_start();
    }
in both places

**Comment by spirit-q2** [comment]
First, make it all work,
Than, try to simplify your work with session. It looks a little bit complicated.

**Comment by whimsicaldreamer** [comment]
@spirit-q2 How do you propose simplifying the work of the session?

**Pull request merged**